<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>546</width>
    <height>605</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>CoDeSys network variable sniffer</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QTabWidget" name="Tabs">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tabRead">
       <attribute name="title">
        <string>Listen</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QGroupBox" name="groupBox">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>175</width>
              <height>75</height>
             </size>
            </property>
            <property name="title">
             <string> Incoming IP</string>
            </property>
            <widget class="QRadioButton" name="anyIP">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>20</y>
               <width>93</width>
               <height>20</height>
              </rect>
             </property>
             <property name="text">
              <string>Any</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
            <widget class="QRadioButton" name="specialIP">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>50</y>
               <width>93</width>
               <height>20</height>
              </rect>
             </property>
             <property name="text">
              <string/>
             </property>
            </widget>
            <widget class="QLineEdit" name="IP">
             <property name="geometry">
              <rect>
               <x>20</x>
               <y>50</y>
               <width>165</width>
               <height>21</height>
              </rect>
             </property>
             <property name="minimumSize">
              <size>
               <width>165</width>
               <height>0</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>165</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="text">
              <string>192.168.1.1</string>
             </property>
             <property name="maxLength">
              <number>15</number>
             </property>
            </widget>
           </widget>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <widget class="QLabel" name="label_3">
              <property name="maximumSize">
               <size>
                <width>16777215</width>
                <height>25</height>
               </size>
              </property>
              <property name="text">
               <string>CobID:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_2">
              <property name="maximumSize">
               <size>
                <width>1345654</width>
                <height>25</height>
               </size>
              </property>
              <property name="text">
               <string>Port:</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_4">
            <item>
             <widget class="QSpinBox" name="CobIdBox">
              <property name="maximumSize">
               <size>
                <width>16777215</width>
                <height>25</height>
               </size>
              </property>
              <property name="maximum">
               <number>65535</number>
              </property>
              <property name="value">
               <number>477</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="portBox">
              <property name="maximumSize">
               <size>
                <width>16777215</width>
                <height>25</height>
               </size>
              </property>
              <property name="maximum">
               <number>65535</number>
              </property>
              <property name="value">
               <number>1202</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_listen">
            <property name="text">
             <string>Listen</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Variable types:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="variables">
          <property name="text">
           <string>REAL,REAL,REAL,REAL,REAL,REAL,REAL,REAL,BYTE</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QPlainTextEdit" name="logBox">
            <property name="plainText">
             <string/>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Separate variables with comma
 i.e. REAL,UINT,SINT

BOOL
BYTE
WORD
DWORD
SINT
USINT
INT
UINT
DINT
UDINT
REAL
LREAL</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <property name="topMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>Log to file</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="checkBox_LogToFile">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="file_prefix">
            <property name="text">
             <string>datafile_prefix</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="comboBox_dataloggerTimestep">
            <item>
             <property name="text">
              <string>Every 1 s</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Every 5 s</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Every 10 s</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Every 1 min</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Every 10 min</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabWrite">
       <attribute name="title">
        <string>Broadcast</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QGroupBox" name="groupBox_2">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>175</width>
              <height>75</height>
             </size>
            </property>
            <property name="title">
             <string> Receiver IP</string>
            </property>
            <widget class="QRadioButton" name="radioButton_broadcast_IP_any">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>20</y>
               <width>102</width>
               <height>20</height>
              </rect>
             </property>
             <property name="text">
              <string>Any</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
            <widget class="QRadioButton" name="radioButton_broadcast_IP_specific">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>50</y>
               <width>102</width>
               <height>20</height>
              </rect>
             </property>
             <property name="text">
              <string/>
             </property>
            </widget>
            <widget class="QLineEdit" name="lineEdit_broadcast_IP">
             <property name="geometry">
              <rect>
               <x>20</x>
               <y>50</y>
               <width>165</width>
               <height>22</height>
              </rect>
             </property>
             <property name="minimumSize">
              <size>
               <width>165</width>
               <height>0</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>165</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="text">
              <string>192.168.255.255</string>
             </property>
            </widget>
           </widget>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_6">
            <item>
             <widget class="QLabel" name="label_6">
              <property name="text">
               <string>CobID</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_7">
              <property name="text">
               <string>Port</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_7">
            <item>
             <widget class="QSpinBox" name="spinBox_broadcast_cobid">
              <property name="maximum">
               <number>9999</number>
              </property>
              <property name="value">
               <number>791</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="spinBox_broadcast_port">
              <property name="maximum">
               <number>9999</number>
              </property>
              <property name="value">
               <number>1202</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_broadcast">
            <property name="text">
             <string>Broadcast</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>Variable types:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="lineEdit_broadcast_variables">
          <property name="text">
           <string>BYTE,BYTE,REAL,REAL</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_9">
          <property name="text">
           <string>Variables:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="lineEdit_broadcast_values">
          <property name="text">
           <string>1,1,2.0,3.0</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_7">
          <property name="topMargin">
           <number>10</number>
          </property>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QCheckBox" name="checkBox_hearBeat">
            <property name="text">
             <string>Heart beat every</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="spinBox_hearBeatInterval">
            <property name="maximum">
             <number>360000</number>
            </property>
            <property name="value">
             <number>500</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_11">
            <property name="text">
             <string>ms in varialbe i=</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="spinBox_hearBeatVariable"/>
          </item>
          <item>
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QWidget" name="widget_heartBeatMarker" native="true">
            <property name="minimumSize">
             <size>
              <width>30</width>
              <height>30</height>
             </size>
            </property>
            <property name="baseSize">
             <size>
              <width>30</width>
              <height>30</height>
             </size>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabMessages">
       <attribute name="title">
        <string>Sniffer</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_8">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <property name="topMargin">
           <number>0</number>
          </property>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_9">
            <property name="topMargin">
             <number>0</number>
            </property>
            <item>
             <widget class="QLabel" name="label_10">
              <property name="text">
               <string>UDP datagram rate:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_12">
              <property name="text">
               <string>Codesys telegram rate:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_13">
              <property name="text">
               <string>Listened telegram rate:</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_10">
            <item>
             <widget class="QLabel" name="label_UDPrate">
              <property name="lineWidth">
               <number>0</number>
              </property>
              <property name="text">
               <string>TextLabel</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_codesysrate">
              <property name="text">
               <string>TextLabel</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_telegramrate">
              <property name="text">
               <string>TextLabel</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_11">
            <item>
             <widget class="QLabel" name="label_16">
              <property name="text">
               <string>1/s</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_17">
              <property name="text">
               <string>1/s</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_18">
              <property name="text">
               <string>1/s</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_12">
            <item>
             <widget class="QLabel" name="label_20">
              <property name="text">
               <string>Update every:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="comboBox_rateTime">
              <item>
               <property name="text">
                <string>1 s</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>5 s</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>10 s</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>1 min</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>10 min</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QPlainTextEdit" name="plainTextEdit_messageLog"/>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_6">
          <property name="topMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QLabel" name="label_19">
            <property name="text">
             <string>Log traffic: </string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="checkBox_logUDPtraffic">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEdit_UDPtrafficPrefix">
            <property name="text">
             <string>datafile_prefix</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="comboBox_UDPtrafficLogCondition">
            <item>
             <property name="text">
              <string>All UPD</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>All Codesys</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Listened</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
